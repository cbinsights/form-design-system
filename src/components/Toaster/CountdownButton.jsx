import React from 'react';
import PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import FDS from 'lib/dictionary/js/styleConstants';
import IconButton from 'components/IconButton';
import DenyIcon from 'lib/icons/react/DenyIcon';

/**
 *
 * @param {Number} circleSize bounding box size of circle in px
 * @param {Number} strokeWidth width of circle stroke
 */
export const getCircleInfo = (circleSize, strokeWidth) => {
  const radius = (circleSize - strokeWidth) / 2;

  return {
    r: radius,
    c: Math.PI * radius * 2,
    centerOffset: circleSize / 2,
  };
};

const CountdownButton = ({ label = 'Close', Icon = DenyIcon, duration, onClick }) => {
  const circleSize = 34;
  const strokeWidth = '2';
  const { r, c, centerOffset } = getCircleInfo(circleSize, strokeWidth);

  const defaultCircleStyle = {
    // strokeDasharray sets up a dashed line in which each dash is equal to
    // the circumference of the circle.
    //
    // We offset the dash by 0 to start to create a full circle
    strokeDasharray: c,
    strokeDashoffset: 0,
    transition: `all ${duration}ms linear`,

    // start the countdown at "midnight" position of circle
    transformOrigin: 'center center',
    transform: 'rotate(-90deg)',
  };

  const transitionStyles = {
    entering: { strokeDashoffset: 0 },

    // offsetting the dash by the exact circle circumference lines up
    // the empty part of the dashed stroke with the circle circumference
    entered: { strokeDashoffset: c },
  };

  return (
    <div
      className="countdownButton"
      style={{
        width: `${circleSize}px`,
        height: `${circleSize}px`,
      }}
    >
      <div className="countdownButton-button">
        <IconButton onClick={onClick} radius="circle" Icon={Icon} label={label} />
      </div>
      <Transition
        in={true}
        appear
        timeout={{
          appear: 1,
          enter: duration,
        }}
        mountOnEnter
      >
        {(state) => (
          <svg
            className="countdownButton-animation"
            width={circleSize}
            height={circleSize}
            viewBox={`0 0 ${circleSize} ${circleSize}`}
          >
            <circle
              cx={centerOffset}
              cy={centerOffset}
              r={r}
              strokeWidth={strokeWidth}
              fill="transparent"
              stroke={FDS.FONT_COLOR_HINT}
              style={{
                ...defaultCircleStyle,
                ...transitionStyles[state],
              }}
            />
          </svg>
        )}
      </Transition>
    </div>
  );
};

CountdownButton.propTypes = {
  /** Duration in ms of countdown animation */
  duration: PropTypes.number.isRequired,

  /** Used to render a FDS Icon (should only be used for FDS Icons, e.g. Icon={DenyIcon}) */
  Icon: PropTypes.func,

  /** adds a label to IconButton */
  label: PropTypes.string,

  /** Click callback for button */
  onClick: PropTypes.func,
};

export default CountdownButton;
