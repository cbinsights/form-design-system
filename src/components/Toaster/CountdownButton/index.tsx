import React from 'react';
import { Transition } from 'react-transition-group';
import FDS from 'dictionary/js/styleConstants';
import IconButton from 'components/IconButton';
import DenyIcon from 'icons/react/DenyIcon';
import Icon from 'components/Icon';
export interface CircleInfo {
  radius: number;
  circumference: number;
  centerOffset: number;
}

/**
 *
 * @param {Number} circleSize bounding box size of circle in px
 * @param {Number} strokeWidth width of circle stroke
 */
export const getCircleInfo = (circleSize: number, strokeWidth: number): CircleInfo => {
  const radius = (circleSize - strokeWidth) / 2;

  return {
    radius,
    circumference: Math.PI * radius * 2,
    centerOffset: circleSize / 2,
  };
};

export interface CountdownButtonProps {
  /** Duration in ms of countdown animation */
  duration: number;

  /** Used to render a FDS Icon (should only be used for FDS Icons, e.g. Icon={DenyIcon}) */
  Icon?: Icon;

  /** adds a label to IconButton */
  label?: string;

  /** Click callback for button */
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const CountdownButton = ({
  label = 'Close',
  Icon = DenyIcon,
  duration,
  onClick,
}: CountdownButtonProps): JSX.Element => {
  const circleSize = 34;
  const strokeWidth = 2;
  const { radius, circumference, centerOffset } = getCircleInfo(circleSize, strokeWidth);

  const defaultCircleStyle = {
    // strokeDasharray sets up a dashed line in which each dash is equal to
    // the circumference of the circle.
    //
    // We offset the dash by 0 to start to create a full circle
    strokeDasharray: circumference,
    strokeDashoffset: 0,
    transition: `all ${duration}ms linear`,

    // start the countdown at "midnight" position of circle
    transformOrigin: 'center center',
    transform: 'rotate(-90deg)',
  };

  const transitionStyles = {
    entering: { strokeDashoffset: 0 },

    // offsetting the dash by the exact circle circumference lines up
    // the empty part of the dashed stroke with the circle circumference
    entered: { strokeDashoffset: circumference },
    exiting: undefined,
    exited: undefined,
    unmounted: undefined,
  };

  return (
    <div
      className="countdownButton"
      style={{
        width: `${circleSize}px`,
        height: `${circleSize}px`,
      }}
    >
      <div className="countdownButton-button">
        <IconButton onClick={onClick} radius="circle" Icon={Icon} label={label} />
      </div>
      <Transition
        in={true}
        appear
        timeout={{
          appear: 1,
          enter: duration,
        }}
        mountOnEnter
      >
        {(state) => (
          <svg
            className="countdownButton-animation"
            width={circleSize}
            height={circleSize}
            viewBox={`0 0 ${circleSize} ${circleSize}`}
          >
            <circle
              cx={centerOffset}
              cy={centerOffset}
              r={radius}
              strokeWidth={strokeWidth}
              fill="transparent"
              stroke={FDS.FONT_COLOR_HINT}
              style={{
                ...defaultCircleStyle,
                ...transitionStyles[state],
              }}
            />
          </svg>
        )}
      </Transition>
    </div>
  );
};

export default CountdownButton;
