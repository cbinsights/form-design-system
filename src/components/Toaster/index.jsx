import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import Toast from './Toast';

export const Toaster = ({
  toastProps = {},
  dismissDelay = 4000,
  isAutoDismiss = true,
  isOpen = false,
  id,
  onDismiss,
}) => {
  useEffect(() => {
    // We need to explicitly check of canDismiss is NOT true, as it has a default
    // parameter of true that this component does not know about
    if (
      isAutoDismiss &&
      toastProps.type !== 'progress' &&
      toastProps.canDismiss !== false
    ) {
      const timer = setTimeout(() => onDismiss(), dismissDelay);

      return function cleanup() {
        clearTimeout(timer);
      };
    }

    return undefined;
  }, [id, isOpen, isAutoDismiss, toastProps.type, toastProps.canDismiss]);

  return (
    <div aria-live="assertive">
      <TransitionGroup>
        {isOpen && (
          <CSSTransition
            key={id}
            appear
            unmountOnExit
            timeout={380}
            classNames="rtgSlideIn"
          >
            <div className="toaster">
              <Toast
                dismissDelay={dismissDelay}
                dismissToast={onDismiss}
                isAutoDismiss={isAutoDismiss}
                content={toastProps.content}
                type={toastProps.type}
                progress={toastProps.progress}
                canDismiss={toastProps.canDismiss}
                actionLabel={toastProps.actionLabel}
                onAction={toastProps.onAction}
                dismissOnAction={toastProps.dismissOnAction}
              />
            </div>
          </CSSTransition>
        )}
      </TransitionGroup>
    </div>
  );
};

const ToasterWrapper = (props) =>
  ReactDOM.createPortal(
    <Toaster {...props} />,
    // eslint-disable-next-line no-undef
    document.body
  );

Toaster.propTypes = {
  /**  A UUID for the toast. Each new toast should have a new UUID. */
  id: PropTypes.string,
  /** Specifies whether Toast is open or not */
  isOpen: PropTypes.bool,
  /** Accepts all props that `Toast` accepts. Refer to Toast component for full documentation */
  toastProps: PropTypes.shape(Toast.propTypes),
  /** Should this toast auto-dismiss itself? */
  isAutoDismiss: PropTypes.bool,
  /** Time in ms to auto-dismiss toast */
  dismissDelay: PropTypes.number,
  /** Callback that is fired when toast is dismissed */
  onDismiss: PropTypes.func,
};

export default ToasterWrapper;
