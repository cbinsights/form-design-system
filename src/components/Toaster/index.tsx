import React, { useEffect } from 'react';
import ReactDOM from 'react-dom';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import Toast, { ToastProps } from './Toast';

const dismissDelay = 4000;

export interface ToasterProps {
  /**  A UUID for the toast. Each new toast should have a new UUID. */
  id?: string;

  /** Specifies whether Toast is open or not */
  isOpen?: boolean;

  /** Should this toast auto-dismiss itself? */
  isAutoDismiss?: boolean;

  onDismiss?: (...args: unknown[]) => void;

  /** Controls the display for your specific toast. */
  toastInstance?: ToastProps;
}

export const Toaster = ({
  isAutoDismiss = true,
  isOpen = false,
  onDismiss,
  toastInstance = {
    content: null,
  },
  id,
}: ToasterProps): JSX.Element => {
  useEffect(() => {
    // We need to explicitly check of canDismiss is NOT true, as it has a default
    // parameter of true that this component does not know about
    if (
      isAutoDismiss &&
      toastInstance.type !== 'progress' &&
      toastInstance.canDismiss !== false &&
      onDismiss
    ) {
      const timer = setTimeout(() => onDismiss(), dismissDelay);

      return function cleanup() {
        clearTimeout(timer);
      };
    }

    return undefined;
  }, [
    id,
    isOpen,
    isAutoDismiss,
    toastInstance.type,
    toastInstance.canDismiss,
    onDismiss,
  ]);

  return (
    <div aria-live="assertive">
      <TransitionGroup>
        {isOpen && (
          <CSSTransition
            key={id}
            appear
            unmountOnExit
            timeout={380}
            classNames="rtgSlideIn"
          >
            <div className="toaster">
              <Toast
                dismissDelay={dismissDelay}
                isAutoDismiss={isAutoDismiss}
                dismissToast={onDismiss}
                content={toastInstance.content}
                type={toastInstance.type}
                progress={toastInstance.progress}
                canDismiss={toastInstance.canDismiss}
                actionLabel={toastInstance.actionLabel}
                onAction={toastInstance.onAction}
                dismissOnAction={toastInstance.dismissOnAction}
              />
            </div>
          </CSSTransition>
        )}
      </TransitionGroup>
    </div>
  );
};

const ToasterWrapper = (props: ToasterProps): React.ReactNode =>
  ReactDOM.createPortal(
    <Toaster {...props} />,
    // eslint-disable-next-line no-undef
    document.body
  );

export default ToasterWrapper;
