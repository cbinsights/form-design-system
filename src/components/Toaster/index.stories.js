import React, { useState } from 'react';
import { v4 } from 'uuid';

import { Toaster } from 'components/Toaster';
import Toast from './Toast';

export const Primary = (args) => {
  const [isToastShowing, setIsToastShowing] = useState(false);
  return (
    <>
      <button onClick={() => setIsToastShowing(!isToastShowing)}>
        {isToastShowing ? 'Hide Toaster' : 'Show Toaster'}
      </button>
      <Toaster
        {...args}
        isOpen={isToastShowing}
        toastInstance={{
          content: 'hey',
          onDismiss: () => setIsToastShowing(false),
          type: 'info',
          progress: 42,
          dismissOnAction: true,
          actionLabel: 'Action',
        }}
      />
    </>
  );
};

export const Playground = () => {
  const defaults = { id: v4(), isOpen: false, isAutoDismiss: true, toastInstance: {} };
  const [params, setParams] = useState(defaults);
  const dismissToast = () => {
    setParams({ ...params, isOpen: false });
  };
  const randomNum = () => Math.floor(Math.random() * 100);

  const { id, isOpen, toastInstance, isAutoDismiss, actionLabel } = params;
  return (
    <div className="padding--all">
      <h4>Core:</h4>
      <button onClick={() => setParams({ ...params, isOpen: true })}>Open Toast</button>
      <button onClick={() => setParams({ ...params, isOpen: false })}>Close Toast</button>
      <button onClick={() => setParams({ ...params, isAutoDismiss: !isAutoDismiss })}>
        Toggle isAutoDismiss {isAutoDismiss ? 'off' : 'on'}
      </button>
      <button onClick={() => setParams({ ...params, id: v4() })}>New ID</button>
      <button onClick={() => setParams({ ...defaults })}>Reset all</button>
      <br />
      <br />
      <h4>Types:</h4>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, type: 'info' },
          })
        }
      >
        info
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, type: 'warn' },
          })
        }
      >
        warn
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, type: 'error' },
          })
        }
      >
        error
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, type: 'success' },
          })
        }
      >
        success
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, type: 'progress' },
          })
        }
      >
        progress
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, progress: randomNum() },
          })
        }
      >
        Add random progress
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, progress: null },
          })
        }
      >
        Remove progress
      </button>
      <br />
      <br />
      <h4>Modify Content:</h4>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: {
              ...params.toastInstance,
              actionLabel: actionLabel === 'Hello' ? 'World' : 'Hello',
            },
          })
        }
      >
        Change Action Label
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, content: 'hello world' },
          })
        }
      >
        Add content
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, content: <b>hello world</b> },
          })
        }
      >
        Add HTML content
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, content: null },
          })
        }
      >
        Remove content
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: { ...params.toastInstance, actionLabel: null },
          })
        }
      >
        Remove Action Label
      </button>
      <br />
      <br />
      <h4>Example usages:</h4>
      <button
        onClick={() =>
          setParams({
            ...params,
            toastInstance: {
              ...params.toastInstance,
              type: 'error',
              content: 'New ID? Yes!',
            },
            id: v4(),
            isOpen: true,
          })
        }
      >
        Error Toast (animate)
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            id: v4(),
            isOpen: true,
            toastInstance: {
              ...params.toastInstance,
              content: 'New ID? Yes!',
              type: 'info',
            },
          })
        }
      >
        Info Toast (animate)
      </button>
      <button
        onClick={() =>
          setParams({
            ...params,
            id,
            isOpen: true,
            toastInstance: {
              ...params.toastInstance,
              content: 'New ID? No.',
              type: 'error',
            },
          })
        }
      >
        Error Toast (do not animate)
      </button>
      <br />
      <br />
      <b>NOTE: this is state is EXTERNAL to toaster and is being passed in</b>
      <pre>{JSON.stringify(params, null, 1)}</pre>
      <Toaster
        id={id}
        isOpen={isOpen}
        isAutoDismiss={isAutoDismiss}
        toastInstance={{
          onDismiss: dismissToast,
          actionLabel: toastInstance.actionLabel,
          progress: toastInstance.progress,
          type: toastInstance.type,
          content: toastInstance.content,
        }}
      />
    </div>
  );
};

Playground.parameters = {
  docs: {
    description: {
      story:
        'This is a playground where you can enable certain functionality and see how Toaster reacts in real time, while being able to visualize the exact prop structure to generate it.',
    },
  },
};

export const Variations = () => (
  <React.Fragment>
    <Toast
      content={
        <p>
          Check out this <strong className="color--heading">toast</strong>
        </p>
      }
      actionLabel="Action"
      onAction={() => {}}
      type="info"
    />
    <br />
    <Toast
      content={
        <p>
          You better <strong className="color--heading">watch out</strong>
        </p>
      }
      actionLabel="run away"
      onAction={() => {}}
      type="warn"
    />
    <br />
    <Toast
      content={
        <p>
          You <strong className="color--heading">Did it!</strong>
        </p>
      }
      type="success"
    />
    <br />
    <Toast
      content={
        <p>
          Failed to <strong className="color--heading">do a thing</strong>
        </p>
      }
      actionLabel="Retry"
      onAction={() => {}}
      type="error"
    />
    <br />
    <Toast
      content={
        <p>
          Uploading a photo of your <strong className="color--heading">your cat</strong>
        </p>
      }
      type="progress"
    />
    <br />
    <Toast
      content={
        <p>
          <strong className="color--heading">42%</strong> complete uploading your cat
          photo
        </p>
      }
      progress={42}
      type="progress"
    />
  </React.Fragment>
);

Variations.parameters = {
  docs: {
    description: {
      story: 'This is merely to give you a better visual idea of all the variations.',
    },
    source: {
      code: 'Hidden. This is merely to give you a visual idea of the themes available.',
    },
  },
};

export default {
  component: Toaster,
  subcomponents: { Toast },
  title: 'components/Toaster',
  parameters: {
    componentSubtitle: 'Displays Toast alerts',
    docs: {
      description: {
        component:
          'Important notes: - **⚠️ Import only one `Toaster` component per application**. - Pass in a brand new `id` (with optional props) if you would like to animate a new toast in. - Passing in new props without changing the id will not retrigger the animation',
      },
    },
  },
};
