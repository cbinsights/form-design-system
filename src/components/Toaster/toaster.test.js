import React from 'react';
import { mount } from 'enzyme';

import Toaster from '.';

describe('Toaster component', () => {

  it('matches snapshot', () => {
    const component = mount(<Toaster isOpen toastProps={{ type: "info", content: <p>test</p> }} />);
    expect(component).toMatchSnapshot();
  });

  it('passes dismiss func to toast', () => {
    const onDismiss = jest.fn()
    const component = mount(<Toaster onDismiss={onDismiss} isOpen toastProps={{ type: "info", content: <p>test</p> }} />);
    const {dismissToast} = component.find("Toast").props();
    expect(typeof dismissToast).toBe('function');
  })

});
