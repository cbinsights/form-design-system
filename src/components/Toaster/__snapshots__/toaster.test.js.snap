// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster component matches snapshot 1`] = `
<ToasterWrapper
  isOpen={true}
  toastProps={
    Object {
      "content": <p>
        test
      </p>,
      "type": "info",
    }
  }
>
  <Portal
    containerInfo={
      <body>
        <div
          aria-live="assertive"
        >
          <div>
            <div
              class="toaster rtgSlideIn-appear rtgSlideIn-appear-active"
            >
              <div
                aria-atomic="true"
                aria-relevant="all"
                class="toast toast--info"
                role="log"
              >
                <div
                  class="toast-bar"
                />
                <div
                  class="toast-content alignChild--left--center"
                >
                  <div
                    class="flex flex--alignStretch flex--row"
                  >
                    <div
                      class="flexItem flexItem--shrink"
                    >
                      <div
                        class="toast-constrainGrowth alignChild--center--center"
                      >
                        <div
                          class="toast-icon media--s"
                        >
                          <div
                            class="fds-icon fds-icon--l"
                            style="fill: #4cabce;"
                          >
                            <svg
                              viewBox="0 0 24 24"
                            >
                              <defs>
                                <path
                                  d="M12,21 C7.02943725,21 3,16.9705627 3,12 C3,7.02943725 7.02943725,3 12,3 C16.9705627,3 21,7.02943725 21,12 C21,16.9705627 16.9705627,21 12,21 Z M12,19 C15.8659932,19 19,15.8659932 19,12 C19,8.13400675 15.8659932,5 12,5 C8.13400675,5 5,8.13400675 5,12 C5,15.8659932 8.13400675,19 12,19 Z M11,11 L13,11 L13,17 L11,17 L11,11 Z M12,10 C11.1715729,10 10.5,9.32842712 10.5,8.5 C10.5,7.67157288 11.1715729,7 12,7 C12.8284271,7 13.5,7.67157288 13.5,8.5 C13.5,9.32842712 12.8284271,10 12,10 Z"
                                  id="prefix__InformationIcona"
                                />
                              </defs>
                              <g
                                fill-rule="evenodd"
                              >
                                <g>
                                  <use
                                    xlink:href="#prefix__InformationIcona"
                                  />
                                </g>
                              </g>
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="flexItem"
                    >
                      <div
                        class="toast-constrainGrowth"
                      >
                        <p>
                          test
                        </p>
                      </div>
                    </div>
                    <div
                      class="flexItem flexItem--shrink"
                    >
                      <div
                        class="toast-constrainGrowth alignChild--center--center"
                      >
                        <div
                          class="countdownButton"
                          style="width: 34px; height: 34px;"
                        >
                          <div
                            class="countdownButton-button"
                          >
                            <button
                              class="fdsIconButton rounded--all border--focus--noTransition transition--default alignChild--center--center fdsIconButton--ghost fdsIconButton--m shape--circle"
                              title="Close"
                            >
                              <div
                                class="fdsIconButton--loading-spinnerWrapper"
                              >
                                <div
                                  class=""
                                />
                              </div>
                              <span
                                class=""
                              >
                                <div
                                  class="fds-icon fds-icon--s"
                                  style="width: 18px; height: 18px;"
                                >
                                  <svg
                                    viewBox="0 0 24 24"
                                  >
                                    <g
                                      fill-rule="evenodd"
                                    >
                                      <g>
                                        <polygon
                                          points="10.363961 11.7781746 4 5.41421356 5.41421356 4 11.7781746 10.363961 18.1421356 4 19.5563492 5.41421356 13.1923882 11.7781746 19.5563492 18.1421356 18.1421356 19.5563492 11.7781746 13.1923882 5.41421356 19.5563492 4 18.1421356"
                                        />
                                      </g>
                                    </g>
                                  </svg>
                                </div>
                              </span>
                            </button>
                          </div>
                          <svg
                            class="countdownButton-animation"
                            height="34"
                            viewBox="0 0 34 34"
                            width="34"
                          >
                            <circle
                              cx="17"
                              cy="17"
                              fill="transparent"
                              r="16"
                              stroke="rgba(64, 64, 64, 0.38)"
                              stroke-width="2"
                              style="stroke-dasharray: 100.53096491487338; stroke-dashoffset: 0; transition: all 4000ms linear; transform-origin: center center; transform: rotate(-90deg);"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    }
  >
    <Toaster
      isOpen={true}
      toastProps={
        Object {
          "content": <p>
            test
          </p>,
          "type": "info",
        }
      }
    >
      <div
        aria-live="assertive"
      >
        <TransitionGroup
          childFactory={[Function]}
          component="div"
        >
          <div>
            <CSSTransition
              appear={true}
              classNames="rtgSlideIn"
              in={true}
              key=".0"
              onExited={[Function]}
              timeout={380}
              unmountOnExit={true}
            >
              <Transition
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={380}
                unmountOnExit={true}
              >
                <div
                  className="toaster"
                >
                  <Toast
                    content={
                      <p>
                        test
                      </p>
                    }
                    dismissDelay={4000}
                    isAutoDismiss={true}
                    type="info"
                  >
                    <div
                      aria-atomic="true"
                      aria-relevant="all"
                      className="toast toast--info"
                      role="log"
                    >
                      <div
                        className="toast-bar"
                      />
                      <div
                        className="toast-content alignChild--left--center"
                      >
                        <Flex>
                          <div
                            className="flex flex--alignStretch flex--row"
                          >
                            <FlexItem
                              shrink={true}
                            >
                              <div
                                className="flexItem flexItem--shrink"
                              >
                                <div
                                  className="toast-constrainGrowth alignChild--center--center"
                                >
                                  <div
                                    className="toast-icon media--s"
                                  >
                                    <InformationIcon
                                      color="#4cabce"
                                      size="l"
                                    >
                                      <WorldIcon
                                        color="#4cabce"
                                        size="l"
                                      >
                                        <div
                                          className="fds-icon fds-icon--l"
                                          style={
                                            Object {
                                              "fill": "#4cabce",
                                              "height": undefined,
                                              "width": undefined,
                                            }
                                          }
                                        >
                                          <InformationIcon>
                                            <svg
                                              viewBox="0 0 24 24"
                                            >
                                              <defs>
                                                <path
                                                  d="M12,21 C7.02943725,21 3,16.9705627 3,12 C3,7.02943725 7.02943725,3 12,3 C16.9705627,3 21,7.02943725 21,12 C21,16.9705627 16.9705627,21 12,21 Z M12,19 C15.8659932,19 19,15.8659932 19,12 C19,8.13400675 15.8659932,5 12,5 C8.13400675,5 5,8.13400675 5,12 C5,15.8659932 8.13400675,19 12,19 Z M11,11 L13,11 L13,17 L11,17 L11,11 Z M12,10 C11.1715729,10 10.5,9.32842712 10.5,8.5 C10.5,7.67157288 11.1715729,7 12,7 C12.8284271,7 13.5,7.67157288 13.5,8.5 C13.5,9.32842712 12.8284271,10 12,10 Z"
                                                  id="prefix__InformationIcona"
                                                />
                                              </defs>
                                              <g
                                                fillRule="evenodd"
                                              >
                                                <g>
                                                  <use
                                                    xlinkHref="#prefix__InformationIcona"
                                                  />
                                                </g>
                                              </g>
                                            </svg>
                                          </InformationIcon>
                                        </div>
                                      </WorldIcon>
                                    </InformationIcon>
                                  </div>
                                </div>
                              </div>
                            </FlexItem>
                            <FlexItem>
                              <div
                                className="flexItem"
                              >
                                <div
                                  className="toast-constrainGrowth"
                                >
                                  <p>
                                    test
                                  </p>
                                </div>
                              </div>
                            </FlexItem>
                            <FlexItem
                              shrink={true}
                            >
                              <div
                                className="flexItem flexItem--shrink"
                              >
                                <div
                                  className="toast-constrainGrowth alignChild--center--center"
                                >
                                  <CountdownButton
                                    duration={4000}
                                    label="Close"
                                  >
                                    <div
                                      className="countdownButton"
                                      style={
                                        Object {
                                          "height": "34px",
                                          "width": "34px",
                                        }
                                      }
                                    >
                                      <div
                                        className="countdownButton-button"
                                      >
                                        <IconButton
                                          Icon={[Function]}
                                          label="Close"
                                          radius="circle"
                                        >
                                          <button
                                            className="fdsIconButton rounded--all border--focus--noTransition transition--default alignChild--center--center fdsIconButton--ghost fdsIconButton--m shape--circle"
                                            title="Close"
                                          >
                                            <div
                                              className="fdsIconButton--loading-spinnerWrapper"
                                            >
                                              <div
                                                className=""
                                              />
                                            </div>
                                            <span
                                              className=""
                                            >
                                              <DenyIcon
                                                customSize={18}
                                              >
                                                <WorldIcon
                                                  customSize={18}
                                                >
                                                  <div
                                                    className="fds-icon fds-icon--s"
                                                    style={
                                                      Object {
                                                        "fill": undefined,
                                                        "height": "18px",
                                                        "width": "18px",
                                                      }
                                                    }
                                                  >
                                                    <DenyIcon>
                                                      <svg
                                                        viewBox="0 0 24 24"
                                                      >
                                                        <g
                                                          fillRule="evenodd"
                                                        >
                                                          <g>
                                                            <polygon
                                                              points="10.363961 11.7781746 4 5.41421356 5.41421356 4 11.7781746 10.363961 18.1421356 4 19.5563492 5.41421356 13.1923882 11.7781746 19.5563492 18.1421356 18.1421356 19.5563492 11.7781746 13.1923882 5.41421356 19.5563492 4 18.1421356"
                                                            />
                                                          </g>
                                                        </g>
                                                      </svg>
                                                    </DenyIcon>
                                                  </div>
                                                </WorldIcon>
                                              </DenyIcon>
                                            </span>
                                          </button>
                                        </IconButton>
                                      </div>
                                      <Transition
                                        appear={true}
                                        enter={true}
                                        exit={true}
                                        in={true}
                                        mountOnEnter={true}
                                        onEnter={[Function]}
                                        onEntered={[Function]}
                                        onEntering={[Function]}
                                        onExit={[Function]}
                                        onExited={[Function]}
                                        onExiting={[Function]}
                                        timeout={
                                          Object {
                                            "appear": 1,
                                            "enter": 4000,
                                          }
                                        }
                                        unmountOnExit={false}
                                      >
                                        <svg
                                          className="countdownButton-animation"
                                          height={34}
                                          viewBox="0 0 34 34"
                                          width={34}
                                        >
                                          <circle
                                            cx={17}
                                            cy={17}
                                            fill="transparent"
                                            r={16}
                                            stroke="rgba(64, 64, 64, 0.38)"
                                            strokeWidth="2"
                                            style={
                                              Object {
                                                "strokeDasharray": 100.53096491487338,
                                                "strokeDashoffset": 0,
                                                "transform": "rotate(-90deg)",
                                                "transformOrigin": "center center",
                                                "transition": "all 4000ms linear",
                                              }
                                            }
                                          />
                                        </svg>
                                      </Transition>
                                    </div>
                                  </CountdownButton>
                                </div>
                              </div>
                            </FlexItem>
                          </div>
                        </Flex>
                      </div>
                    </div>
                  </Toast>
                </div>
              </Transition>
            </CSSTransition>
          </div>
        </TransitionGroup>
      </div>
    </Toaster>
  </Portal>
</ToasterWrapper>
`;
